fontforge/ufo.c

```diff
@@ -1086,8 +1103,38 @@ static int UFOOutputFontInfo(const char *basedir, SplineFont *sf, int layer) {
 /* Same keys in both formats */
     PListAddString(dictnode,"familyName",sf->familyname_with_timestamp ? sf->familyname_with_timestamp : sf->familyname);
     PListAddString(dictnode,"styleName",SFGetModifiers(sf));
-    if (sf->pfminfo.os2_family_name != NULL) PListAddString(dictnode,"styleMapFamilyName", sf->pfminfo.os2_family_name);
-    if (sf->pfminfo.os2_style_name != NULL) PListAddString(dictnode,"styleMapStyleName", sf->pfminfo.os2_style_name);
+    {
+        char* preferredFamilyName = fetchTTFAttribute(sf,ttf_preffamilyname);
+        char* preferredSubfamilyName = fetchTTFAttribute(sf,ttf_prefmodifiers);
+        char* styleMapFamily;
+        if (sf->styleMapFamilyName != NULL) {
+            /* Empty styleMapStyleName means we imported a UFO that does not have this field. Bypass the fallback. */
+            if (sf->styleMapFamilyName[0]!='\0')
```

Comment:
I would point out that this would fail to round-trip a blank name, but I think that a name cannot be blank.